### ISSUE TITLE
EXCEPTION

### ISSUE THREAD

**Issue by Justman10000** [owncloud/music#1080]
Hi, got this error:

> OC\IntegrityCheck\Exceptions\MissingSignatureException
> Signature data not found.

**Comment by paulijar** [comment]
So what were you doing when you got this?

**Comment by Justman10000** [comment]
> So what were you doing when you got this?

```bash
mkdir apps/music
cd apps/music
git init
git remote add origin https://github.com/owncloud/music.git
git pull origin master
git branch -m master
rm -r .* *.md COPYING LICENSE
```

**Comment by Justman10000** [comment]
> So what were you doing when you got this?

Have installed this so:

```bash
mkdir apps/music
cd apps/music
git init
git remote add origin https://github.com/owncloud/music.git
git pull origin master
git branch -m master
rm -r .* *.md COPYING LICENSE
```

**Comment by paulijar** [comment]
Are you planning to do development on this app, or do you want to use bleeding edge software which hasn't been released yet? Because for most users, using the pre-packaged deliveries should be the preferred option of installing. So either install it from the app store or download the tar.gz from https://github.com/owncloud/music/releases and extract it to the `apps` folder.

If you pull the project yourself from github, then obviously the application will not be signed and I assume that your error message is now somehow related to that. However, not having signature shouldn't cause any other problems except for the nagging on the admin UI.

Also, when pulling the project from github, you need to build the front-end bundle by running these commands under `apps/music`:
```
npm install --deps
npm run build
```
And for that, you need to have `npm` installed.

**Comment by Justman10000** [comment]
> If you pull the project yourself from github, then obviously the application will not be signed
Can I sign this myself?

**Comment by Justman10000** [comment]
> If you pull the project yourself from github, then obviously the application will not be signed

Can I sign this myself?

**Comment by paulijar** [comment]
I don't believe there is any self-signing system, at least officially. If I remember correctly, I obtained the certificates used for signing years ago by registering to some services maintained by ownCloud and Nextcloud. I think the system was such that one application ID may be registered only once. That is, the application ID `music` is now permanently bound to a private key in my possession, and only I can create signed deliveries of this app. Hence, whenever a user sees a signed `music` application delivery, she can be reasonably certain that it is from me and not from some (possibly malicious) 3rd party.

But personally, I'm using an unsigned version of Music app 99% of the time. I only sign the app whenever I'm making a release. It would be cumbersome to sign it after each code change.

**Comment by Justman10000** [comment]
> But personally, I'm using an unsigned version of Music app 99% of the time. I only sign the app whenever I'm making a release. It would be cumbersome to sign it after each code change.

But then the same error occurs as with me?

**Comment by paulijar** [comment]
I haven't seen this one. So is it actually blocking you from using the app, or what does it cause, in addition to showing the error? Which version of ownCloud or Nextcloud are you using?

In case this blocks the application installation, would it work better if you first install the Music app from the app store, and then replace the folder `apps/music` with the one you pulled and built yourself?
